src/solver/optimisation/opt_appel_solveur_lineaire.cpp

```diff
@@ -94,20 +94,34 @@ class TimeMeasurement
     clock::time_point start_;
     clock::time_point end_;
 };
-
+char** VectorOfStringToCharPP(std::vector<std::string>& in, std::vector<char*>& pointerVec)
+{
+    std::transform(in.begin(),
+                   in.end(),
+                   std::back_inserter(pointerVec),
+                   [](std::string& str) { return str.empty() ? nullptr : str.data(); });
+    return pointerVec.data();
+}
 bool OPT_AppelDuSimplexe(PROBLEME_HEBDO* problemeHebdo,
                          int NumIntervalle,
                          const int optimizationNumber,
                          std::shared_ptr<OptPeriodStringGenerator> optPeriodStringGenerator)
 {
     PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre = problemeHebdo->ProblemeAResoudre;
-    Optimization::PROBLEME_SIMPLEXE_NOMME Probleme(ProblemeAResoudre->NomDesVariables,
-                                                   ProblemeAResoudre->NomDesContraintes,
+    std::vector<char*> pointerVecVar;
+    std::vector<char*> pointerVecCnt;
+    auto NomsDesVariablesCharPP
+      = VectorOfStringToCharPP(ProblemeAResoudre->NomDesVariables, pointerVecVar);
+    auto NomDesContraintesCharPP
+      = VectorOfStringToCharPP(ProblemeAResoudre->NomDesContraintes, pointerVecCnt);
+    Optimization::PROBLEME_SIMPLEXE_NOMME Probleme(NomsDesVariablesCharPP,
```

Comment:
In order to remain valid, this instance of `PROBLEME_SIMPLEXE_NOMME` implicitly needs `pointerVecVar` and `pointerVecCnt`. If one of these vectors is destroyed, `NomsDesVariablesCharPP` or `NomDesContraintesCharPP` will become dangling pointers, leading to UB.

For this reason, I suggest including `pointerVecCnt` & `pointerVecVar` as private members of class `Optimization::PROBLEME_SIMPLEXE_NOMME`. This will tie the lifetime of these objects so the outside world doesn't need to worry about this.

So you can just write something like
```cpp
    Optimization::PROBLEME_SIMPLEXE_NOMME Probleme(ProblemeAResoudre->NomDesVariables,
ProblemeAResoudre->NomDesContraintes);
```
and hide the implementation details in the constructor of `Optimization::PROBLEME_SIMPLEXE_NOMME`.

---

There's another layer of hidden dependency, which is `ProblemeAResoudre->NomDesVariables` & `ProblemeAResoudre->NomDesContraintes`, for which we have exactly the same problem. But since this instance of `PROBLEME_ANTARES_A_RESOUDRE` has a "long life" (created at the start, destroyed at the end), it's not really an issue for now.